multiqc/multiqc.py

```diff
@@ -362,33 +395,195 @@ def run(
 
+    # Throw an error if we are using an unsupported version of Python
+    if sys.version_info < tuple(map(int, OLDEST_SUPPORTED_PYTHON_VERSION.split("."))):
+        return RunResult(
+            message="You are running MultiQC with Python {}. "
+            "Please upgrade Python! MultiQC does not support Python < {}, "
+            "things will break.".format(sys.version_info, OLDEST_SUPPORTED_PYTHON_VERSION),
+            sys_exit_code=1,
+        )
+
+    console = _set_up_logging(verbose, quiet, no_ansi)
+
+    console.print(
+        f"\n  [dark_orange]///[/] [bold][link=https://multiqc.info]MultiQC[/link][/] :mag: [dim]| v{config.version}\n"
+    )
+    logger.debug(f"This is MultiQC v{config.version}")
+
+    _init(
+        analysis_dir=analysis_dir,
+        dirs=dirs,
+        dirs_depth=dirs_depth,
+        no_clean_sname=no_clean_sname,
+        title=title,
+        report_comment=report_comment,
+        template=template,
+        module=module,
+        require_logs=require_logs,
+        exclude=exclude,
+        outdir=outdir,
+        ignore=ignore,
+        ignore_samples=ignore_samples,
+        use_filename_as_sample_name=use_filename_as_sample_name,
+        replace_names=replace_names,
+        sample_names=sample_names,
+        sample_filters=sample_filters,
+        file_list=file_list,
+        filename=filename,
+        make_data_dir=make_data_dir,
+        no_data_dir=no_data_dir,
+        data_format=data_format,
+        zip_data_dir=zip_data_dir,
+        force=force,
+        ignore_symlinks=ignore_symlinks,
+        no_report=no_report,
+        export_plots=export_plots,
+        plots_flat=plots_flat,
+        plots_interactive=plots_interactive,
+        strict=strict,
+        lint=lint,
+        development=development,
+        make_pdf=make_pdf,
+        no_megaqc_upload=no_megaqc_upload,
+        config_file=config_file,
+        cl_config=cl_config,
+        profile_runtime=profile_runtime,
+        no_ansi=no_ansi,
+        custom_css_files=custom_css_files,
+        **kwargs,
+    )
+
+    try:
+        run_modules, run_module_names = _file_search(
+            file_list=file_list,
+            ignore=ignore,
+            ignore_samples=ignore_samples,
+        )
+
+        # Create the temporary working directories
+        tmp_dir = tempfile.mkdtemp()
+        logger.debug(f"Using temporary directory for creating report: {tmp_dir}")
+
+        # Load the template
+        template_mod = config.avail_templates[config.template].load()
+
+        _run_modules(
+            tmp_dir=tmp_dir,
+            template_mod=template_mod,
+            filename=filename,
+            run_modules=run_modules,
+            run_module_names=run_module_names,
+        )
+
+        _general_stats_table()
+
+        _export_sources()
+
+        _write_json_dump()
+
+        _write_html_and_data(
+            tmp_dir=tmp_dir,
+            template_mod=template_mod,
+            filename=filename,
+        )
```

Comment:
Love how clear this is! ideally `multiqc.py` can end up being split up into functions in all the relevant sub-files, and this block can go into `__main__.py` or somewhere, and we can finally get rid of `multiqc.multiqc` megascript :)